### PR TITLE
Update info file comparison reflect new version of Atum 0.2.6.

### LINKED ISSUE #67: Update Info File Comparison to reflect Atum enchantments
## Background
Atum is going to have some work done. Upgrade Info File Comparison to reflect these.

## Feature
- [Atum #23](https://github.com/AbsaOSS/atum/issues/23)
- [Atum #19](https://github.com/AbsaOSS/atum/issues/19)

### PR DESCRIPTION
**Pull request by HuvarVer**
Closes #67

### CODE REVIEW

File: infoFileComparison/src/main/scala/za/co/absa/hermes/infoFileComparison/AtumModelUtils.scala [hunk 1]
```diff
@@ -134,6 +134,8 @@ object AtumModelUtils {
       val diffs =
         simpleCompare(checkpoint.name, otherCheckpoint.name, s"$curPath.name") ::
           simpleCompare(checkpoint.workflowName, otherCheckpoint.workflowName, s"$curPath.workflowName") ::
+          simpleCompare(checkpoint.software, otherCheckpoint.software, s"$curPath.software") ::
+          simpleCompare(checkpoint.version, otherCheckpoint.version, s"$curPath.version") ::
```
Comment (Zejnilovic) [inline infoFileComparison/src/main/scala/za/co/absa/hermes/infoFileComparison/AtumModelUtils.scala]: I think this is a leftover from previous path taken

File: infoFileComparison/src/main/scala/za/co/absa/hermes/infoFileComparison/AtumModelUtils.scala [hunk 2]
```diff
@@ -142,8 +144,23 @@ object AtumModelUtils {
...
+      val wasVersion = checkpoint.version.getOrElse("Not Specified")
+      val isVersion = otherCheckpoint.version.getOrElse("Not Specified")
+
+      logCheckpointDiff(wasSoftware, wasVersion, isSoftware, isVersion)
+
       diffs.flatten ::: controls
+
     }
+
+    private def logCheckpointDiff (refSoftware: String, refVersion: String,
```
Comment (Zejnilovic) [inline infoFileComparison/src/main/scala/za/co/absa/hermes/infoFileComparison/AtumModelUtils.scala]: Please add a path of the checkpoint to the log

File: infoFileComparison/src/main/scala/za/co/absa/hermes/infoFileComparison/AtumModelUtils.scala [hunk 3]
```diff
@@ -142,8 +144,23 @@ object AtumModelUtils {
...
+                                   newSoftware: String, newVersion: String): Unit = {
+      if (refSoftware != newSoftware) scribe.info(s"Software was - $refSoftware and is - $newSoftware.")
```
Comment (Zejnilovic) [inline infoFileComparison/src/main/scala/za/co/absa/hermes/infoFileComparison/AtumModelUtils.scala]: - is the message enough? try to imaging seeing "Software was - alfa and is - beta" in the middle of all logs.
- I think it should be a warning. A warning would tell me as a user that something isn't 100% but it doesn't have to be an issue.
- I would use only one `if` and print both if any differ. Would make for a more concise message.

File: infoFileComparison/src/main/scala/za/co/absa/hermes/infoFileComparison/AtumModelUtils.scala [hunk 4]
```diff
@@ -142,8 +142,26 @@ object AtumModelUtils {
...
+
     }
+
+    private def logCheckpointDiff (refSoftware: String, newSoftware: String,
+                                   refVersion: String, newVersion: String,
+                                   curPath: String): Unit = {
+      if (refSoftware != newSoftware || refVersion != newVersion)
+        scribe.warn(
+          s"""Software that generated checkpoints in $curPath differs.
+             |Was - $refSoftware($refVersion) Is - $newSoftware($newVersion)""".stripMargin)
```
Comment (Zejnilovic) [inline infoFileComparison/src/main/scala/za/co/absa/hermes/infoFileComparison/AtumModelUtils.scala]: last nit: I would either have both `was` and `is` on their own line or strip the new lines `.replaceAll("[\\r\\n]", "")`

Comment (HuvarVer) [reply infoFileComparison/src/main/scala/za/co/absa/hermes/infoFileComparison/AtumModelUtils.scala]: 👍

**Review by Zejnilovic — APPROVED** [review]

**Pull request merged**